.github/workflows/macos_tests.yml

```diff
@@ -0,0 +1,82 @@
+name: MacOS tests
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+  release:
+  workflow_dispatch:
+
+jobs:
+  tests:
+    name: Test - ${{ matrix.os }} - Build type ${{ matrix.build_type }} - Build testing ${{ matrix.build_testing }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - macos-11
+          - macos-12
+          # TODO: can change to macos-latest (but it's currently aliased to macos-11)
+        build_type:
+          - Release
+        build_testing:
+          - ON
+          - OFF
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Setup cache
+        id: cache-test-data
+        uses: actions/cache@v3
+        with:
+          path: ${{ github.workspace }}/tdms/tests/system/data
+          key: test_data
+
+      - name: Set up MATLAB
+        uses: matlab-actions/setup-matlab@v1
+
+      - name: Install dependencies
+        run: |
+          brew install fftw llvm libomp python3
```

Comment:
Is `llvm` strictly necessary?  I thought that installing libomp would suffice, but I haven't tried myself.

Reply:
Huh. [You're right](https://github.com/samcunliffe/TDMS/runs/8300142473?check_suite_focus=true).

Reply:
Oh. I needed to do it on my local M2 machine because: architectures.